api4/team.go

```diff
@@ -499,14 +499,14 @@ func addUserToTeamFromInvite(c *Context, w http.ResponseWriter, r *http.Request)
 	var member *model.TeamMember
 	var err *model.AppError
 
-	if c.App.Session.Props[model.SESSION_PROP_IS_GUEST] == "true" {
-		c.Err = model.NewAppError("addUserToTeamFromInvite", "api.team.add_user_to_team_from_invite.guest.app_error", nil, "", http.StatusForbidden)
-		return
-	}
-
 	if len(tokenId) > 0 {
 		member, err = c.App.AddTeamMemberByToken(c.App.Session.UserId, tokenId)
 	} else if len(inviteId) > 0 {
+		if c.App.Session.Props[model.SESSION_PROP_IS_GUEST] == "true" {
```

Comment:
Moving this here because I only want to disallow the usage of team invite link, not the token based invites (which is the previous branch in the if).